File: src/com/sorting/subSequence.java
package com.sorting;

public class subSequence {
    public static String[] findSubSequence(String str){


        if(str.length() == 0){
            String[] ans = {""};
            return ans;
        }
        String[] smallAns = findSubSequence(str.substring(1));
        String[] ans = new String[2*smallAns.length];

        for(int i = 0;i<smallAns.length;i++){
            ans[i] = smallAns[i];
        }

        for(int i = 0; i< smallAns.length;i++){
        ans[i+smallAns.length] = str.charAt(0)+smallAns[i];
        }

        return ans;
    }
    public static void main(String[] args) {
        String[] subsequence = findSubSequence("surya");
        for(int i = 0; i<subsequence.length;i++){
            System.out.println(subsequence[i]);
        }
    }
}

File: src/com/sorting/subSequence.js
const subSequence = (str) =>{
    if(str.length == 0){
        return [""];
    }

    let smallAns = subSequence(str.substring(1));
    let newArr = [];

    for(let i = 0;i<smallAns.length;i++){
     newArr[i] = smallAns[i];
    }

    for(let i = 0; i<str.length;i++){
        newArr[i+ smallAns.length] = str[0].concat(smallAns[i]);
    }
    return newArr;
}

let output = subSequence("surya");

for(let i = 0; i<output.length;i++){
    console.log(output[i]);
}
